### PR TITLE
make PumpLL idempotent to reduce cpu use a bit

### PR DESCRIPTION
**Pull request by majestrate**
in order to reduce excess cpu use in PathContext's `Pump{Up,Down}stream()` make the only place in the code that calls those functions idempotent (`Router::PumpLL`).

### CODE REVIEW

File: llarp/handlers/exit.cpp [hunk 1]
```diff
@@ -837,18 +809,14 @@ namespace llarp
...
     {
-      auto range = m_ActiveExits.equal_range(ep->PubKey());
-      auto itr = range.first;
-      while (itr != range.second)
+      for (auto [itr, end] = m_ActiveExits.equal_range(ep->PubKey()); itr != end; ++itr)
       {
         if (itr->second->GetCurrentPath() == ep->GetCurrentPath())
         {
-          itr = m_ActiveExits.erase(itr);
+          m_ActiveExits.erase(itr);
```
Comment (majestrate) [inline llarp/handlers/exit.cpp:816]: this erase will change the thing we are iterating over if it rehashes

Comment (jagerman) [reply llarp/handlers/exit.cpp:816]: Yes, but as the next line says right before it returns, "// now ep is gone af"

File: llarp/handlers/exit.cpp [hunk 2]
```diff
@@ -677,19 +655,14 @@ namespace llarp
...
+        return nullptr;
+      if (auto itr = m_ActiveExits.find(pk); itr != m_ActiveExits.end())
       {
-        auto itr = m_ActiveExits.find(pk);
-        if (itr != m_ActiveExits.end())
-        {
-          if (itr->second->PubKey() == pk)
-            endpoint = itr->second.get();
-        }
+        if (itr->second->PubKey() == pk)
```
Comment (tewinget) [inline llarp/handlers/exit.cpp:664]: this `if` seems redundant with the above changes.

Comment (tewinget) [reply llarp/handlers/exit.cpp:664]: actually it seems it may have been redundant before as well?

File: llarp/router/router.cpp [hunk 3]
```diff
@@ -69,13 +69,30 @@ namespace llarp
     _running.store(false);
     _lastTick = llarp::time_now_ms();
     m_NextExploreAt = Clock_t::now();
+    m_Pump = _loop->make_waker([this]() {
+      paths.PumpDownstream();
```
Comment (majestrate) [inline llarp/router/router.cpp]: move these two to PumpLL

**Review by tewinget — APPROVED** [review]

**Pull request merged**